src/api/app/controllers/staging_project/staged_requests_controller.rb

```diff
@@ -1,5 +1,59 @@
 class StagingProject::StagedRequestsController < StagingProjectController
+  before_action :set_staging_workflow, :set_project, :set_xml_hash, only: [:create, :destroy]
+
   def index
-    @requests = @project.staged_requests
+    @requests = @staging_project.staged_requests
+  end
+
+  def create
+    authorize @staging_project, :update?
+    requests = @staging_workflow.unassigned_requests.where(id: @xml_hash[:number])
+
+    if @staging_project.staged_requests << requests
+      StageRequestsJob.perform_later(@staging_project, requests.pluck(:id), User.current)
+      render_ok
+    else
+      render_error(
+        status: 422,
+        errorcode: 'invalid_request',
+        message: "Could not unassign requests to #{@staging_project}: #{@staging_project.errors.full_messages.to_sentence}"
+      )
+    end
+  end
+
+  def destroy
+    authorize @staging_project, :update?
+    requests = @staging_project.staged_requests.where(id: @xml_hash[:number])
+
+    if @staging_project.staged_requests.delete(requests)
+      UnstageRequestsJob.perform_later(@staging_project, requests.pluck(:id), User.current)
+      render_ok
+    else
+      render_error(
+        status: 422,
+        errorcode: 'invalid_request',
+        message: "Could not unassign request from #{@staging_project}: #{@staging_project.errors.full_messages.to_sentence}"
+      )
+    end
+  end
+
+  private
+
+  def set_xml_hash
+    @xml_hash = (Xmlhash.parse(request.body.read) || {}).with_indifferent_access
```

Comment:
Because Xmlhash uses only strings, if you do ``@hash[:number]`` it will be empty for instance. Params are always with_different_access in Rails so we should also use it otherwise it can cause weird side effects...